app/components/post_list/post_list.android.js

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {StyleSheet} from 'react-native';
+import RecyclerViewList, {DataSource, RecyclerRefreshControl} from 'react-native-recyclerview-list';
+
+import {ListTypes} from 'app/constants';
+
+import PostListBase from './post_list_base';
+
+const SCROLL_UP_MULTIPLIER = 3.5;
+
+export default class PostList extends PostListBase {
+    constructor(props) {
+        super(props);
+
+        this.contentOffsetY = 0;
+        this.contentHeight = 0;
+
+        this.state = {
+            refreshing: false,
+            managedConfig: {},
+            dataSource: new DataSource(props.postIds, this.keyExtractor),
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.channelId !== nextProps.channelId) {
+            this.contentOffsetY = 0;
+            this.contentHeight = 0;
+        }
+
+        if (nextProps.postIds !== this.props.postIds && this.mounted) {
```

Comment:
Do we need to check `this.mounted` here? Unless we're calling this manually, lifecycle methods usually aren't called after unmount